### ISSUE TITLE
Why is 

### ISSUE THREAD

**Issue by nachiketrss** [dotnet/AspNetCore.Docs#24618]
This is regarding the contents in the following 2 sections:

1. https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/linux-apache?view=aspnetcore-6.0#publish-and-copy-over-the-app
2. https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/linux-apache?view=aspnetcore-6.0#https-configuration

The `Properties/launchSettings.json` file is a development only file (as mentioned here: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-6.0#development-and-launchsettingsjson). 

So why does this document (which pertains to deployment) mention about changing the value in this file please?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cfd9a1f0-c1b6-4443-0700-4c5beafeed03
* Version Independent ID: c22ca7a0-d1d4-200d-fc36-a9798d350c6e
* Content: [Host ASP.NET Core on Linux with Apache](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/linux-apache?view=aspnetcore-6.0#publish-and-copy-over-the-app)
* Content Source: [aspnetcore/host-and-deploy/linux-apache.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/host-and-deploy/linux-apache.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **shboyer**

**Comment by guardrex** [comment]
Hello @nachiketrss ... This is just to call out that running locally and securely may require additional configuration. The proxy is insecure, so the app can run that way in production with the server config set up to handle HTTPS with clients. It's true that the default in the app's `launchSettings.json` file is for both HTTP and HTTPS. However ... depending on the host ... one might need to take additional steps to run the app securely locally. It's the same deal with macOS scenarios. We try to at least raise awareness. Sometimes, we just cross-link out for coverage. In this case, the doc author decided to place some guidance right here in the topic. For more info, see ...

https://docs.microsoft.com/aspnet/core/host-and-deploy/proxy-load-balancer

**Comment by nachiketrss** [comment]
This particular line is very confusing to me- `If the app is run locally and isn't configured to make secure connections (HTTPS), adopt either of the following approaches:`

The 2nd point under this line doesn't add any security - does it?

**Comment by guardrex** [comment]
Right ... it's just suggesting that you can run it insecurely locally. You can remove the HTTPS address from the config. Doesn't it launch the first one? I don't recall of the top of my head. You could run a quick local test ... `dotnet run` an app with the HTTPS endpoint first ... then run it again with the HTTP endpoint first. See if in each case it tries to run the app with **_the first address listed_**. If so, then the 2nd bullet makes sense ... to run locally insecurely with the browser opening the right endpoint, kill 🔪 the HTTPS address (or swap them).

**Comment by nachiketrss** [comment]
1. In production the app isn't meant to be run using dotnet run command. Instead it is run with something like `dotnet appname.dll` or `dotnet appname.exe`. This way, the settings in the launchSettings.json file doesn't get picked up, as it is meant only for `dotnet run` (running in dev mode).

2. Even assuming I use `dotnet run`, and when testing on a browser, it is not enough to remove the url from the launchSettings.json. Because there is `app.UseHttpsRedirection();` in program.cs which redirects to https.

**Comment by guardrex** [comment]
1. The text states that "If the app is run locally ...". That's all that this pertains to.
2. Yes. We should mention deactivating HTTPS Redirection Middleware in the Development environment for this scenario. We can cross-link to https://docs.microsoft.com/aspnet/core/fundamentals/environments#configure-services-and-middleware-by-environment (`xref:fundamentals/environments#configure-services-and-middleware-by-environment`). Given that the example there isn't explicit for this scenario, the code is short, and the code probably won't churn later (🤞), we could show it for this bullet point ...

   ```csharp
   if (!app.Environment.IsDevelopment())
   {
       app.UseHttpsRedirection();
   }
   ```

Also, I just confirmed that reversing the order of the URLs in `applicationUrl` so that the insecure endpoint is first does spawn the browser at the insecure endpoint ... it uses the first one. Still tho, I'd recommend leaving this text in its current form to *remove the address*. The point of this bullet is that the dev has no intention of testing locally at the secure endpoint.

@Rick-Anderson ... These updates will only take a few minutes to enact. Assign me if you wish, and I'll knock it out on Tuesday.

**Comment by nachiketrss** [comment]
I agree with your point 1. But think of it this way: I come to this page to deploy the web app/api. For me locally doesn't necessarily mean development. It could mean locally in the sense that the app runs locally behind a reverse proxy like nginx. Here the word locally is bit ambiguous.

**Comment by guardrex** [comment]
Let's flesh that out here, but you'll probably see that "locally" is used on its own quite a bit across the docs. I'll drop a draft copy-'n-paste here ...

> If the app is run locally in the \[Development environment](xref:fundamentals/environments) and isn't configured to make secure connections (HTTPS), adopt either of the following approaches:

... and mentioning in passing that ...

* The content of the Linux-Nginx article is versioned, so earlier versions probably need the update, too.
* The Linux-Apache topic should have the same improvements: https://docs.microsoft.com/aspnet/core/host-and-deploy/linux-apache#publish-and-copy-over-the-app
